foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/PythonLintingMigration.java

```diff
@@ -0,0 +1,153 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import com.boozallen.aissemble.upgrade.util.pom.LocationAwareMavenReader;
+import com.boozallen.aissemble.upgrade.util.pom.PomModifications;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.model.Build;
+import org.apache.maven.model.InputLocation;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.boozallen.aissemble.upgrade.util.pom.PomHelper.getLocationAnnotatedModel;
+import static com.boozallen.aissemble.upgrade.util.pom.PomHelper.writeModifications;
+
+/**
+ * This migration disables failing on Python linting for existing projects.
+ */
+public class PythonLintingMigration extends AbstractAissembleMigration {
+    private Plugin getHabushuPlugin(Model model) {
+        List<Plugin> plugins = model.getBuild().getPluginManagement().getPlugins();
+        for (Plugin plugin : plugins) {
+            if ("habushu-maven-plugin".equalsIgnoreCase(plugin.getArtifactId())) {
+                return plugin;
+            }
+        }
+        return null;
+    }
+
+    private Object getConfiguration(Plugin plugin) {
+        Object lintingConfiguration = plugin.getConfiguration();
+        return lintingConfiguration;
+    }
+
+    private boolean hasLinting(Object configuration) {
+        String configurationString = configuration.toString();
+        if (configurationString.contains("sourceFailOnLintErrors") || configurationString.contains("testFailOnLintErrors")) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        if (!"com.boozallen.aissemble".equalsIgnoreCase(model.getParent().getGroupId())) {
```

Comment:
C: I don't think this is correct... The downstream project will likely have a totally different group ID than this and will still need linting failure disabled.  Maybe a good substitute check would be `"habushu".equals(model.getPackaging())`, but not sure what you were going for here.

Reply:
Update. I understand now we're only running this on the root project pom. So this is detecting that the parent is the build-parent.  That makes sense.  You might pull this logic out into a method to make it a bit more clear. Something like `isRootProjectPom`.

Reply:
Refactored the code to pull logic into a method. Thanks Emily!